### ISSUE TITLE
Usage string should not use main when calling a script

### ISSUE THREAD

**Issue by jmgilman** [nushell/nushell#7126]
### Describe the bug

When executing a script and passing the help flag (`nu myscript.nu --help`), the usage string utilizes `main` as the executable. 

```
$ nu build.nu --help

Builds an image

Usage:
  > main <image>    # <-----------------  Uses main

Flags:
  -h, --help - Display the help message for this command

Parameters:
  image <String>: The name of the image to build
```

### How to reproduce

1. Create a script using a `main` function
2. Run `nu myscript.nu --help`

### Expected behavior

For the uninitiated, the usage of `main` as the executable can be confusing. It would be more natural to include the name of the script:

```
$ nu build.nu --help

Builds an image

Usage:
  > build <image>    # <-----------------  Use script name by default

Flags:
  -h, --help - Display the help message for this command

Parameters:
  image <String>: The name of the image to build
```

### Configuration

N/A

**Comment by rgwood** [comment]
Hmm, I'm a little confused by your expected behaviour.

I can see how the current output is confusing, but given that you're running it like `nu build.nu`, suggesting that the user run it like `build <image>` is not right either, no?

What about `nu build.nu`?

```
$ nu build.nu --help

Builds an image

Usage:
  > nu build.nu <image>
...
```

Or even `nu <script path>`?
```
$ nu build.nu --help

Builds an image

Usage:
  > nu <script path> <image>
...
```

**Comment by jmgilman** [comment]
Sorry, that was an oversight on my part. Yes, it should be `build.nu` in my example case. Whether or not the script path is included is a good question -- I would posit in most cases that most autogenerated help text does not include the full path.

**Comment by jmgilman** [comment]
Looking at some [cursory SO answers](https://stackoverflow.com/questions/5474732/how-can-i-add-a-help-method-to-a-shell-script), I'm fairly certain the UNIX standard is to use just the base name of the script in the usage section. 

> I still think that's a little off, because the user can't actually run the script like that. I'm thinking it would make more sense to suggest nu build.nu instead of just build.nu, what do you think?

It depends on the contents of the script. I would expect in most cases that a shebang would be provided, in which case it's more than possible to execute the script without invoking `nu` directly.

**Comment by rgwood** [comment]
Shebangs are only available on *nix OSs and we don't have a guarantee that the script contains one (although I guess we could check).

**Comment by jmgilman** [comment]
True. I'll leave it up to the core team. My experience has been the usage line contains only the base name of the script and/or binary. 

I suppose a (potentially more difficult) alternative is to let the script author control what gets put there. They already have some level of control using comments.

**Comment by rgwood** [comment]
On a related note, we probably want to clean up the error message when giving a script the wrong parameters.
![image](https://user-images.githubusercontent.com/26268125/202350903-8bf6c685-279b-43b6-817b-489d08dba0e8.png)

**Comment by rgwood** [comment]
Spent a bit of time looking at this and it's sadly not quite trivial to fix. At the point where the help content is generated, the engine is not aware of the script file - it's just handling a call to a command named `main`, as far as it's concerned.

Might come back to this another time, but if someone else wants to pick it up here are the relevant bits:

https://github.com/nushell/nushell/blob/8e4b85e29ba34a96a9d80ae63ac97d2f1fd3f0b4/crates/nu-cli/src/eval_file.rs#L39-L53

https://github.com/nushell/nushell/blob/8e4b85e29ba34a96a9d80ae63ac97d2f1fd3f0b4/crates/nu-engine/src/eval.rs#L41-L46

**Comment by jmgilman** [comment]
Yes, that was the same conclusion I had drawn earlier. Piping the context of the script name all the way to where the help is generated is not a straightforward task, unfortunately.

**Comment by jmgilman** [comment]
> On a related note, we probably want to clean up the error message when giving a script the wrong parameters. ![image](https://user-images.githubusercontent.com/26268125/202350903-8bf6c685-279b-43b6-817b-489d08dba0e8.png)

Yes, I agree with this, I'm not usually a fan of spilling out the guts of the script because of a missing parameter. What parameter is missing and maybe displaying the usage line again should be sufficient.

**Comment by weirdan** [comment]
This will likely be fixed when #13465 lands
